src/App.tsx

```diff
@@ -15,36 +16,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState<string[]>(goodsFromServer);
+  const [sortedBy, setSortedBy] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortAlphabetically = () => {
+    if (isReversed) {
+      setGoodsList([...goodsList].sort().reverse());
+    } else {
+      setGoodsList([...goodsList].sort());
+    }
+
+    setSortedBy(SortType.Alphabetically);
+  };
+
+  const sortByLength = () => {
+    if (isReversed) {
+      setGoodsList([...goodsList].sort((a, b) => b.length - a.length));
+    } else {
+      setGoodsList([...goodsList].sort((a, b) => a.length - b.length));
+    }
+
+    setSortedBy(SortType.Length);
+  };
+
+  const reversed = () => {
```

Comment:
The naming is not perfect - you should call it with a verb

```suggestion
  const reverse = () => {
```

```suggestion
  const handleReverse = () => {
```

Same goes for `reseted` function